astropy/visualization/lupton_rgb.py

```diff
@@ -0,0 +1,469 @@
+                len(pedestal)
+            except TypeError:
+                pedestal = 3*[pedestal]
+
+            if len(pedestal) != 3:
+                raise ValueError("please provide 1 or 3 pedestals.")
+
+            image = list(image)        # needs to be mutable
+            for i, im in enumerate(image):
+                if pedestal[i] != 0.0:
+                    image[i] = im - pedestal[i]  # n.b. a copy
+        else:
+            pedestal = len(image)*[0.0]
+
+        image = compute_intensity(*image)
+
+        zscale_limits = ZScaleInterval().get_limits(image)
+        zscale = LinearMapping(*zscale_limits, image=image)
+        data_range = zscale.maximum - zscale.minimum[0]  # zscale.minimum is always a triple
+        minimum = zscale.minimum
+
+        for i, level in enumerate(pedestal):
+            minimum[i] += level
+
+        AsinhMapping.__init__(self, minimum, data_range, Q)
+        self._image = image
+
+
+def make_lupton_rgb(image_r, image_g=None, image_b=None, minimum=0, data_range=5, Q=8,
+                    saturated_border_width=0, saturated_pixel_value=None,
+                    x_size=None, y_size=None, rescale=None,
+                    filename=None):
+    """
+    Return a Red/Green/Blue color image from up to 3 images using an asinh stretch.
+    The input images can be int or float, and in any range or bit-depth.
+
+    For a more detailed look at the use of this method, see the document
+    :ref:`astropy-visualization-rgb`.
+
+    Parameters
+    ----------
+
+    image_r : `~numpy.ndarray`
+        Image to map to red (if None, use the image passed to the constructor).
+    image_g : `~numpy.ndarray`
+        Image to map to green (if None, use image_r).
+    image_b : `~numpy.ndarray`
+        Image to map to blue (if None, use image_r).
+    minimum : float
+        Intensity that should be mapped to black (a scalar or array for R, G, B).
+    data_range : float
+        minimum+data_range defines the white level of the image.
+    Q : float
+        The asinh softening parameter.
+    saturated_border_width : int
+        If saturated_border_width is non-zero, replace saturated pixels with saturated_pixel_value.
+    saturated_pixel_value : float
+        Value to replace saturated pixels with.
+    x_size : int
+        Desired width of RGB image (or None).  If y_size is None, preserve aspect ratio.
+    y_size : int
+        Desired height of RGB image (or None).
+    rescale : float
+        Make size of output image rescale*size of the input image.
+        Cannot be specified if x_size or y_size are given.
+    filename: str
+        Write the resulting RGB image to a file (file type determined frome extension).
+
+    Returns
+    -------
+    rgb : `~numpy.ndarray`
+        RGB (integer, 8-bits per channel) color image as an NxNx3 numpy array.
+    """
+    image_r = image_r
+    if image_g is None:
+        image_g = image_r
+    if image_b is None:
+        image_b = image_r
```

Comment:
Not sure if it makes sense to use the red image if others are not provided. As the goal is to make a RGB image, the user should provide the three images (and can always give 2 or 3 times the same one). So this could be removed I think.